File: frontend/src/app/model/MedicalCard/medical-card.ts
import {Patient} from "../patient/patient";
import { v4 as uuidv4 } from 'uuid';

export class MedicalCard {
  id: string;
  condition: string;
  symptoms: any;
  patient: object;
  admission: string;
  dateTime: any;
  doctor: object;
  isHospitalized: boolean;
  bed: number;
  created_at:any;
  updated_at:any;

  constructor(
    condition: string,
    patient: object,
    admission: string,
    doctor: object,
    dateTime: any,

  ) {
    this.condition = condition;
    this.patient = patient;
    this.admission = admission;
    this.dateTime = dateTime;
    this.doctor = doctor;
  }


}

File: src/main/java/com/example/hospital/model/MedicalCard.java
package com.example.hospital.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.jpa.convert.threeten.Jsr310JpaConverters;

import java.sql.Date;
import java.sql.Time;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.Set;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "medical_card")
public class MedicalCard {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String condition;
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "medical_history_id", referencedColumnName = "id")
    private MedicalHistory medicalHistory;
    @ManyToOne
    private Patient patient;
    private String admission;
    private LocalDateTime dateTime;
    private Date treatmentCancelledDate;
    private boolean treatmentCancelled;
    @ManyToOne
    private Doctor doctor;
    private boolean hospitalized;
    @OneToOne(optional = true)
    private Bed bed;

    @OneToMany
    private Set<Diary> diarySet;
    private boolean historyFilled;

    private Date createdAt;
    private Date updatedAt;

}
